lib/dev/kit/system.rb

```diff
@@ -142,6 +145,25 @@ def system(*a, sudo: false, env: ENV, **kwargs)
           $CHILD_STATUS
         end
 
+        # Split off trailing partial UTF-8 Characters. UTF-8 Multibyte characters start with a 11xxxxxx byte that tells
+        # how many following bytes are part of this character, followed by some number of 10xxxxxx bytes.  This simple
+        # algorithm will split off a whole trailing multi-byte character.
+        def split_partial_characters(data)
+          last_byte = data.getbyte(-1)
+          return [data, ''] if (last_byte & 0b1000_0000).zero?
+
+          # UTF-8 is up to 6 characters per rune, so we could never want to trim more than that, and we want to avoid
+          # allocating an array for the whole of data with bytes
+          min_bound = -[6, data.bytesize].min
+          final_bytes = data.byteslice(min_bound..-1).bytes
+          partial_character_sub_index = final_bytes.rindex { |byte| byte & 0b1100_0000 == 0b1100_0000 }
```

Comment:
So if we had, e.g.,

```
1100...
0000...
1000...
1000...
```

we would (kind of incorrectly) trim the whole sequence. That's probably fine, it would be rare that we would ever get output not formatted as UTF-8 and I'm not sure it's worth additional complexity to do better.